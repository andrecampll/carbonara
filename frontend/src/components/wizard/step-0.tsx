import { Button, Typography } from '@mui/material'
import Image from 'next/image'
import { useWizard } from 'react-use-wizard'

import { Step } from './step'

export function Step0() {
  const { previousStep, nextStep } = useWizard()

  return (
    <Step>
      <div className="flex flex-col gap-4">
        <header>
          <Typography variant="h5">
            Lets Calculate Your Carbon Footprint
          </Typography>
          <Typography fontWeight={400}>
            Follow the steps to calculate your carbon footprint. We will provide
            you with a detailed report at the end. Lets get started!
          </Typography>
        </header>

        <div className="my-4 flex flex-col items-center gap-4">
          <Image
            alt=""
            width={200}
            height={200}
            src="/images/carbon-light.webp"
          />
        </div>

        <footer className="flex w-full items-center justify-between">
          <Button onClick={previousStep} color="secondary" type="submit">
            Cancel
          </Button>
          <Button
            onClick={nextStep}
            variant="contained"
            color="secondary"
            type="submit"
          >
            Next
          </Button>
        </footer>
      </div>
    </Step>
  )
}
