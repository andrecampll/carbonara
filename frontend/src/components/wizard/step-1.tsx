import { zodResolver } from '@hookform/resolvers/zod'
import { Button, Typography } from '@mui/material'
import Image from 'next/image'
import { useCallback } from 'react'
import { useForm } from 'react-hook-form'
import { useWizard } from 'react-use-wizard'
import { z } from 'zod'

import { Input } from '../ui/input'
import { useFootprintWizard } from './hooks/use-footprint-wizard'

const footprintStep1Schema = z.object({
  electrictyUsage: z.string().min(1, {
    message: 'Electricity usage is required',
  }),
})

type FootprintWizardFormStep1Data = z.infer<typeof footprintStep1Schema>

export function Step1() {
  const { previousStep, nextStep } = useWizard()
  const { updateWizard, wizard } = useFootprintWizard()

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FootprintWizardFormStep1Data>({
    resolver: zodResolver(footprintStep1Schema),
    defaultValues: {
      electrictyUsage: wizard?.electrictyUsage?.toString() ?? '',
    },
  })

  const onSubmit = useCallback(
    async (formData: FootprintWizardFormStep1Data) => {
      // handle submit
      updateWizard({
        electrictyUsage: Number(formData.electrictyUsage),
      })
      nextStep()
    },
    [nextStep, updateWizard],
  )

  return (
    <div className="flex flex-col gap-4">
      <header>
        <Typography variant="h5">Electricity Usage</Typography>
        <Typography fontWeight={400}>
          Enter your monthly electricity usage in kilowatt hours (Kwh) to
          calculate your carbon footprint.
        </Typography>
      </header>

      <form
        onSubmit={handleSubmit(onSubmit)}
        className="my-4 flex flex-col items-center gap-4"
      >
        <Image alt="" width={300} height={300} src="/images/electricity.webp" />

        <Input
          label="Electricity Usage (Kwh/Month):"
          type="number"
          helperText={errors.electrictyUsage?.message}
          color="secondary"
          {...register('electrictyUsage')}
        />

        <footer className="flex w-full items-center justify-between">
          <Button onClick={previousStep} color="secondary">
            Back
          </Button>
          <Button variant="contained" color="secondary" type="submit">
            Next
          </Button>
        </footer>
      </form>
    </div>
  )
}
