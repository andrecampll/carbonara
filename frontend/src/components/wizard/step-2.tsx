import { zodResolver } from '@hookform/resolvers/zod'
import { Button, Typography } from '@mui/material'
import Image from 'next/image'
import { useCallback } from 'react'
import { useForm } from 'react-hook-form'
import { useWizard } from 'react-use-wizard'
import { z } from 'zod'

import { Input } from '../ui/input'
import { useFootprintWizard } from './hooks/use-footprint-wizard'
import { Step } from './step'

const footprintStep2Schema = z.object({
  transportationUsage: z.string().min(1, {
    message: 'Transportation gasoline usage is required',
  }),
})

type FootprintWizardFormStep2Data = z.infer<typeof footprintStep2Schema>

export function Step2() {
  const { previousStep, nextStep } = useWizard()
  const { updateWizard, wizard } = useFootprintWizard()

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FootprintWizardFormStep2Data>({
    resolver: zodResolver(footprintStep2Schema),
    defaultValues: {
      transportationUsage: wizard?.transportationUsage?.toString() ?? '',
    },
  })

  const onSubmit = useCallback(
    async (formData: FootprintWizardFormStep2Data) => {
      updateWizard({
        transportationUsage: Number(formData.transportationUsage),
      })
      nextStep()
    },
    [nextStep, updateWizard],
  )

  return (
    <Step>
      <div className="flex flex-col gap-4">
        <header>
          <Typography variant="h5">Transportation Gasoline Usage</Typography>
          <Typography fontWeight={400}>
            Enter your monthly transportation gasoline usage in gallons to
            calculate your carbon footprint.
          </Typography>
        </header>

        <form
          onSubmit={handleSubmit(onSubmit)}
          className="my-4 flex flex-col items-center gap-4"
        >
          <Image alt="" width={300} height={300} src="/images/gas.png" />

          <Input
            label="Transportation Gasoline Usage (Gallons/Month):"
            type="number"
            helperText={errors.transportationUsage?.message}
            {...register('transportationUsage')}
          />

          <footer className="flex w-full items-center justify-between">
            <Button onClick={previousStep} color="secondary">
              Back
            </Button>
            <Button variant="contained" color="secondary" type="submit">
              Next
            </Button>
          </footer>
        </form>
      </div>
    </Step>
  )
}
