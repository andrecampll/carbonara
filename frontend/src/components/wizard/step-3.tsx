import { zodResolver } from '@hookform/resolvers/zod'
import { Button, Typography } from '@mui/material'
import Image from 'next/image'
import { useCallback } from 'react'
import { useForm } from 'react-hook-form'
import { useWizard } from 'react-use-wizard'
import { z } from 'zod'

import { Input } from '../ui/input'
import { useFootprintWizard } from './hooks/use-footprint-wizard'

const footprintStep3Schema = z.object({
  shortFlights: z.string().min(1, {
    message: 'Short flights is required',
  }),
  mediumFlights: z.string().min(1, {
    message: 'Medium flights is required',
  }),
  longFlights: z.string().min(1, {
    message: 'Long flights is required',
  }),
})

type FootprintWizardFormStep3Data = z.infer<typeof footprintStep3Schema>

export function Step3() {
  const { previousStep, nextStep } = useWizard()
  const { updateWizard, wizard } = useFootprintWizard()

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FootprintWizardFormStep3Data>({
    resolver: zodResolver(footprintStep3Schema),
    defaultValues: {
      shortFlights: wizard?.shortFlights?.toString() ?? '',
      mediumFlights: wizard?.mediumFlights?.toString() ?? '',
      longFlights: wizard?.longFlights?.toString() ?? '',
    },
  })

  const onSubmit = useCallback(
    async (formData: FootprintWizardFormStep3Data) => {
      // handle submit
      updateWizard({
        shortFlights: Number(formData.shortFlights),
        mediumFlights: Number(formData.mediumFlights),
        longFlights: Number(formData.longFlights),
      })
      nextStep()
    },
    [nextStep, updateWizard],
  )

  return (
    <div className="flex flex-col gap-4">
      <header>
        <Typography variant="h5">Airplane Flights</Typography>
        <Typography fontWeight={400}>
          Enter how many short, medium, and long flights you take per year to
          calculate your carbon footprint.
        </Typography>
      </header>

      <form
        onSubmit={handleSubmit(onSubmit)}
        className="my-4 flex flex-col items-center gap-4"
      >
        <Image alt="" width={300} height={300} src="/images/airplane.webp" />

        <Input
          label="Short Flights (4 hours):"
          type="number"
          helperText={errors.shortFlights?.message}
          color="secondary"
          {...register('shortFlights')}
        />
        <Input
          label="Medium Flights (6 hours):"
          type="number"
          helperText={errors.mediumFlights?.message}
          color="secondary"
          {...register('mediumFlights')}
        />
        <Input
          label="Long Flights (8 hours):"
          type="number"
          helperText={errors.longFlights?.message}
          color="secondary"
          {...register('longFlights')}
        />

        <footer className="flex w-full items-center justify-between">
          <Button onClick={previousStep} color="secondary">
            Back
          </Button>
          <Button variant="contained" color="secondary" type="submit">
            Next
          </Button>
        </footer>
      </form>
    </div>
  )
}
