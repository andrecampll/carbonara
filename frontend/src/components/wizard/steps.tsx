'use client'

import { useMutation } from '@apollo/client'
import { useCallback } from 'react'
import { Wizard } from 'react-use-wizard'

import {
  CalculateFootprintMutationInput,
  Footprint,
} from '@/data/types/footprint'
import { CALCULATE_FOOTPRINT_MUTATION } from '@/graphql/mutations/calculateFootprint'
import { useFootprint } from '@/hooks/use-footprint'

import { Step0 } from './step-0'
import { Step1 } from './step-1'
import { Step2 } from './step-2'
import { Step3 } from './step-3'
import { Step4 } from './step-4'

type CalculateFootprintMutationResponse = {
  calculate: {
    footprint: Footprint
  }
}

export function Steps() {
  const { addFootprint } = useFootprint()

  const [calculateFunction] = useMutation<CalculateFootprintMutationResponse>(
    CALCULATE_FOOTPRINT_MUTATION,
  )

  const handleCalculateFootprint = useCallback(
    async (wizardData: CalculateFootprintMutationInput['input']) => {
      const { data: footPrintData } = await calculateFunction({
        variables: {
          input: wizardData,
        },
      })

      addFootprint(footPrintData?.calculate.footprint as Footprint)
    },
    [calculateFunction, addFootprint],
  )

  return (
    <Wizard>
      <Step0 />
      <Step1 />
      <Step2 />
      <Step3 />
      <Step4 onFinalSubmit={handleCalculateFootprint} />
    </Wizard>
  )
}
