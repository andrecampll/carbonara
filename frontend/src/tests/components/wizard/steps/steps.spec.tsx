import { cleanup, render, screen } from '@testing-library/react'
import { Wizard } from 'react-use-wizard'
import { afterEach, describe, expect, it } from 'vitest'

import { Step0 } from '../../../../components/wizard/steps/step-0'
import { Step1 } from '../../../../components/wizard/steps/step-1'
import { Step2 } from '../../../../components/wizard/steps/step-2'
import { Step3 } from '../../../../components/wizard/steps/step-3'
import { Step4 } from '../../../../components/wizard/steps/step-4'

afterEach(cleanup)

describe('Steps', () => {
  it('should be able to render Step0 component', () => {
    const { container } = render(
      <Wizard>
        <Step0 />
      </Wizard>,
    )

    expect(
      screen.getByText(/Lets Calculate Your Carbon Footprint/i),
    ).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render Step1 component', () => {
    const { container } = render(
      <Wizard>
        <Step1 />
      </Wizard>,
    )

    expect(screen.getByText(/Enter your monthly electricity/i)).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render Step2 component', () => {
    const { container } = render(
      <Wizard>
        <Step2 />
      </Wizard>,
    )

    expect(
      screen.getByText(/Enter your monthly transportation gasoline usage/i),
    ).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render Step3 component', () => {
    const { container } = render(
      <Wizard>
        <Step3 />
      </Wizard>,
    )

    expect(screen.getByText(/Airplane Flights/i)).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render Step4 component', () => {
    const { container } = render(
      <Wizard>
        <Step4 onFinalSubmit={async () => {}} />
      </Wizard>,
    )

    expect(
      screen.getByText(
        /Select your dietary choice to calculate your carbon footprint/i,
      ),
    ).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })
})
