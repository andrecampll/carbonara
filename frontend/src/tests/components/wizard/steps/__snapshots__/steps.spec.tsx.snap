// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Steps > should be able to render Step0 component 1`] = `
<div
  style="opacity: 0; transform: translateX(800px) translateZ(0);"
>
  <div
    class="flex flex-col gap-4"
  >
    <header>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Lets Calculate Your Carbon Footprint
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
      >
        Follow the steps to calculate your carbon footprint. We will provide
      you with a detailed report at the end. Lets get started!
      </p>
    </header>
    <div
      class="my-4 flex flex-col items-center gap-4"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fcarbon-light.webp&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fcarbon-light.webp&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fcarbon-light.webp&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </div>
    <footer
      class="flex w-full items-center justify-end"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Next
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </footer>
  </div>
</div>
`;

exports[`Steps > should be able to render Step1 component 1`] = `
<div
  style="opacity: 0; transform: translateX(800px) translateZ(0);"
>
  <div
    class="flex flex-col gap-4"
  >
    <header>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Electricity Usage
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
      >
        Enter your monthly electricity usage in kilowatt hours (Kwh) to
      calculate your carbon footprint.
      </p>
    </header>
    <form
      class="my-4 flex flex-col items-center gap-4"
      id="electricity-usage-form"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Felectricity.webp&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Felectricity.webp&w=384&q=75 1x, /_next/image?url=%2Fimages%2Felectricity.webp&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Electricity Usage (Kwh/Month):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="electrictyUsage"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Electricity Usage (Kwh/Month):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <footer
      class="flex w-full items-center justify-between"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        form="electricity-usage-form"
        tabindex="0"
        type="submit"
      >
        Next
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </footer>
  </div>
</div>
`;

exports[`Steps > should be able to render Step2 component 1`] = `
<div
  style="opacity: 0; transform: translateX(800px) translateZ(0);"
>
  <div
    class="flex flex-col gap-4"
  >
    <header>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Transportation Gasoline Usage
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
      >
        Enter your monthly transportation gasoline usage in gallons to
      calculate your carbon footprint.
      </p>
    </header>
    <form
      class="my-4 flex flex-col items-center gap-4"
      id="transportation-usage-form"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fgas.png&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fgas.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fgas.png&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Transportation Gasoline Usage (Gallons/Month):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            name="transportationUsage"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Transportation Gasoline Usage (Gallons/Month):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <footer
      class="flex w-full items-center justify-between"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        form="transportation-usage-form"
        tabindex="0"
        type="submit"
      >
        Next
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </footer>
  </div>
</div>
`;

exports[`Steps > should be able to render Step3 component 1`] = `
<div
  style="opacity: 0; transform: translateX(800px) translateZ(0);"
>
  <div
    class="flex flex-col gap-4"
  >
    <header>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Airplane Flights
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
      >
        Enter how many short, medium, and long flights you take per year to
      calculate your carbon footprint.
      </p>
    </header>
    <form
      class="my-4 flex flex-col items-center gap-4"
      id="airplane-flights-form"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fairplane.webp&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fairplane.webp&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fairplane.webp&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r2:"
          id=":r2:-label"
        >
          Short Flights (4 hours):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            name="shortFlights"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Short Flights (4 hours):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          Medium Flights (6 hours):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            name="mediumFlights"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Medium Flights (6 hours):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r4:"
          id=":r4:-label"
        >
          Long Flights (8 hours):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r4:"
            name="longFlights"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Long Flights (8 hours):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <footer
      class="flex w-full items-center justify-between"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        form="airplane-flights-form"
        tabindex="0"
        type="submit"
      >
        Next
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </footer>
  </div>
</div>
`;

exports[`Steps > should be able to render Step4 component 1`] = `
<div
  style="opacity: 0; transform: translateX(800px) translateZ(0);"
>
  <div
    class="flex flex-col gap-4"
  >
    <header>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Dietary Choice
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
      >
        Select your dietary choice to calculate your carbon footprint.
      </p>
    </header>
    <form
      class="my-4 flex flex-col items-center gap-4"
      id="dietary-choice-form"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fmeat.png&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fmeat.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fmeat.png&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-5r88sx-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r5:"
            id=":r5:-label"
          >
            Dietary Choice:
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wnwmuc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r5:"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Dietary Choice:
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </form>
    <footer
      class="flex w-full items-center justify-between"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        form="dietary-choice-form"
        tabindex="0"
        type="submit"
      >
        Next
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </footer>
  </div>
</div>
`;
